common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/BlockTransferMessage.java

```diff
@@ -42,7 +43,7 @@
   /** Preceding every serialized message is its type, which allows us to deserialize it. */
   public enum Type {
     OPEN_BLOCKS(0), UPLOAD_BLOCK(1), REGISTER_EXECUTOR(2), STREAM_HANDLE(3), REGISTER_DRIVER(4),
-    HEARTBEAT(5);
+    HEARTBEAT(5), REGISTER_KUBERNETES_APP(6);
```

Comment:
I think REGISTER_DRIVER was added specifically for mesos; because I can't seem to find references outside of Mesos code. Do you think we should share it? My only concern would be that changes to it might break us inadvertently.